app/matches.py

```diff
@@ -15,45 +15,54 @@ class Matches(db.Model):
   date_added = db.Column(db.Date())
   last_edit = db.Column(db.Date())
 
+
   def __repr__(self):
     return '<Matches %r>' % self.id
   
+
   def findById(id):
       match = Matches.query.get(id)
       if (match is None):
         return None
       else:
         return match.serialize()
 
-  def update(id, event, players, winners, score, category):
+
+  def update(id, event, players, score, category):
       if (id is None):
-        raise Exception('fields cannot be null')
+        raise Exception('id cannot be null')
       
       match = Matches.query.get(id)
 
       if (event is not None):
+        Matches.validateEvent(event)
         match.event = event
-        db.session.commit()
+
       if (players is not None):
+        Matches.validatePlayers(players)
         match.players = players
-        db.session.commit()
-      if (winners is not None):
-        match.winners = winners
-        db.session.commit()
+
       if (score is not None):
-        match.score = score
-        db.session.commit()
+        match.score = json.loads(score)
+        Matches.validateScore(match.score)
+        match.winner = Matches.getWinner(score, players, event)
+
       if (category is not None):
         match.category = category
-        db.session.commit()
-      match.last_edit = datetime.today().strftime('%Y-%m-%d-%H:%M:%S')
 
+      match.last_edit = datetime.today()
+
+      Matches.validatePlayersAndEvents(match.players, match.event)
+
+      db.session.commit()
       return 'success', 201
```

Comment:
Let's keep the HTTP logic in the `routes.py` file. How about we remove this line here, and then in `route.py` we return this message?
```suggestion
```